borgmatic/hooks/snapper.py

```diff
@@ -0,0 +1,137 @@
+        exclude_dirs = set(map(Path, hook_config['exclude']))
+        snapper_dirs -= exclude_dirs
+
+    processed_dirs = set()
+    altered_dirs = set()
+
+    for snapper_dir in snapper_dirs:
+        if snapper_dir not in _available_configs():
+            msg = (
+                f'Source directory "{snapper_dir}" was configured to use its latest snapper snapshot for backup, '
+                f'but a corresponding snapper config could not be found'
+            )
+            if include_all:
+                logger.warning(msg)
+                continue
+            else:
+                raise ValueError(msg)
+        config = _available_configs()[snapper_dir]
+        available_snapshots = _call_snapper('-c', config, 'list', '--disable-used-space')
+        latest_snapshot_number = str(available_snapshots[config][-1]['number'])
+        new_src_dir = snapper_dir / '.snapshots' / latest_snapshot_number / 'snapshot'
+
+        if not new_src_dir.exists():
+            msg = (
+                f'Detected snapshot number {latest_snapshot_number} to be the latest for '
+                f'source directory {snapper_dir}, but the deduced directory ({new_src_dir}) is not present. '
+                f'Likely causes are .snapshots not being mounted properly or no snapshots have been taken yet'
+            )
+            raise ValueError(msg)
+        processed_dirs.add(snapper_dir)
+        altered_dirs.add(new_src_dir)
+    return list(map(str, altered_dirs.union(source_directories - processed_dirs)))
+
+
+def fix_extracted_dirs(hook_config, log_prefix, source_directories, destination_path):
+    '''
+    Renames extracted configured source_directories from snapper snapshot path back to their original pre-snapshot path
+    Example: /some/path/.snapshots/3/snapshot to /some/path
+    '''
+
+    if not source_directories:
+        logger.warning(
+            f'{log_prefix}: No source_directories configured. Unable to rename snapshot directories. '
+            f'Please restore source_directories config to be same, when creating the archive for best '
+            f'results'
+        )
+        return
+    source_directories = set(map(Path, source_directories))
+    destination_path = Path(destination_path) if destination_path else Path(os.getcwd())
+
+    if hook_config['include'] == 'all':
+        snapper_dirs = copy(source_directories)
+    else:
+        include_dirs = set(map(Path, hook_config['include']))
+        snapper_dirs = source_directories.intersection(include_dirs)
+
+    if hook_config.get('exclude'):
+        exclude_dirs = set(map(Path, hook_config['exclude']))
+        snapper_dirs -= exclude_dirs
+
+    for snapper_dir in snapper_dirs:
+        # remove leading slash to allow joining with other paths
+
+        if snapper_dir.is_absolute():
+            snapper_dir = Path(str(snapper_dir)[1:])
+        dest_snapper_dir = destination_path / snapper_dir
+        snap_dir = dest_snapper_dir / '.snapshots'
```

Comment:
Variable naming between `snapper_dir` and `snap_dir` is a little confusing to me. Reading how they're defined, I can see the difference between them. But especially in a function/loop with so many different directory variables, more distinct variable naming would help a ton.